### PR TITLE
`--erase-all` to erase all nonvolatile memory

### LINKED ISSUE #368: Erase-all flash option
Would an option to erase all flash be considered generally useful? The gen 3 nRF52 boards have AP Protect enabled by default, so it might be handy for an optional  erase all to enable subsequent flashing.

### PR DESCRIPTION
**Pull request by Urhengulas**
Fixes #368

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -62,10 +62,11 @@ fn run_target_program(elf_path: &Path, chip_name: &str, opts: &cli::Opts) -> any
     let probe = probe::open(opts)?;
 
     let probe_target = target_info.probe_target.clone();
+    let permissions = Permissions::default().allow_erase_all();
```
Comment (jonathanpallant) [inline src/main.rs:65]: Could you avoid obtaining this permission if you are not going to erase all?

**Review by jonathanpallant — CHANGES_REQUESTED** [review]
Left a comment about permissions.

File: src/main.rs (hunk 1 above)
Comment (Urhengulas) [reply src/main.rs]: Yes, makes sense!

**Pull request merged**